charts/spire/charts/spire-server/templates/_helpers.tpl

```diff
@@ -123,10 +123,9 @@ Create the name of the service account to use
 {{- $lst = append $lst $entry }}
 {{- end }}
 {{- end }}
-{{- if gt (len $lst) 0 }}
+{{- $lst = append $lst "parseTime=true" }}
```

Comment:
Maybe  pipe it through uniq to ensure that users that have set it already dont get a second one?

Reply:
Added the `uniq` but still see the duplicate slip through every 4th or so time i run `helm template`:

```
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true\u0026parseTime=true",
```

Reply:
Google says `\u0026 is the unicode representation of AMPERSAND.` Not sure why it would be doing that though...

Reply:
Maybe thats fine....

Do you have any tarballs in the charts/spire/charts/ directory? Seems similar to the issue marco was having with inconsistency?

Reply:
There is a Chart.lock file but no tarballs

Reply:
> Added the uniq but still see the duplicate slip through every 4th or so time i run helm template:

```
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true",
➜  helm-charts git:(2182) ✗ helm template charts/spire | grep connection_string
                "connection_string": "root:${DBPW}@tcp(mysql.default.svc:3306)/spire?parseTime=true\u0026parseTime=true",
```

How is that getting ran? Not seeing a values file or raw values specified.

I'm seeing it consistently work...
```
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ helm template charts/spire -f examples/external-mysql/values.yaml | grep connection_string
                "connection_string": "spire:${DBPW}@tcp(mysql:3306)/spire?parseTime=true",
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ more examples/external-mysql/
README.md    values.yaml  
(ruamel2) [kfox@zathras spiffe-helm-charts-3]$ more examples/external-mysql/values.yaml 
spire-server:
  dataStore:
    sql:
      databaseType: mysql
      databaseName: spire
      host: mysql
      port: 3306
      username: spire
```

So, the patch does seem to work for me.